src/wallet/wallet.cpp

```diff
@@ -981,6 +981,45 @@ bool CWallet::LoadToWallet(const uint256& hash, const UpdateWalletTxFn& fill_wtx
     return true;
 }
 
+void CWallet::RemoveTransactions(const CScript& script)
+{
+    AssertLockHeld(cs_wallet);
```

Comment:
After the rebase, I'm getting compile warnings for this line consistent with the CI:
```
wallet/wallet.cpp:986:5: warning: calling function 'AssertLockHeldInternal<AnnotatedMixin<std::__1::recursive_mutex> >' requires holding mutex 'cs_wallet' exclusively [-Wthread-safety-analysis]
    AssertLockHeld(cs_wallet);
    ^
./sync.h:81:28: note: expanded from macro 'AssertLockHeld'
#define AssertLockHeld(cs) AssertLockHeldInternal(#cs, __FILE__, __LINE__, &cs)
                           ^
wallet/wallet.cpp:989:54: warning: reading variable 'mapWallet' requires holding mutex 'cs_wallet' [-Wthread-safety-analysis]
    for (std::map<uint256, CWalletTx>::iterator it = mapWallet.begin(); it != mapWallet.end(); ++it) {
                                                     ^
wallet/wallet.cpp:989:79: warning: reading variable 'mapWallet' requires holding mutex 'cs_wallet' [-Wthread-safety-analysis]
    for (std::map<uint256, CWalletTx>::iterator it = mapWallet.begin(); it != mapWallet.end(); ++it) {
                                                                              ^
wallet/wallet.cpp:1001:67: warning: reading variable 'mapWallet' requires holding mutex 'cs_wallet' [-Wthread-safety-analysis]
                std::map<uint256, CWalletTx>::const_iterator mi = mapWallet.find(txin.prevout.hash);
                                                                  ^
wallet/wallet.cpp:1002:27: warning: reading variable 'mapWallet' requires holding mutex 'cs_wallet' [-Wthread-safety-analysis]
                if (mi != mapWallet.end()) {
                          ^
wallet/wallet.cpp:1011:56: warning: calling function 'IsMine' requires holding mutex 'cs_wallet' exclusively [-Wthread-safety-analysis]
        if (txContainsScript && !IsFromMe(*wTx.tx) && !IsMine(*wTx.tx)) {
                                                       ^
wallet/wallet.cpp:1017:9: warning: reading variable 'mapWallet' requires holding mutex 'cs_wallet' [-Wthread-safety-analysis]
        mapWallet.erase(wtx->GetHash());
        ^
7 warnings generated.
```

Reply:
Wasn't able to reproduce myself but seems I was able to fix it